src/ncp/ncp_base.cpp

```diff
@@ -1212,6 +1212,19 @@ otError NcpBase::CommandHandler_RESET(uint8_t aHeader)
 
     otError error = OT_ERROR_NONE;
 
+#if OPENTHREAD_RADIO
+    mInstance->RadioStackReset();
```

Comment:
One advantage of original RESET is that it is able to recover from software errors. 
What if a platform still want to use `otInstanceReset` for `RESET` command?

Should we add a optional argument to `RESET` command to specify the reset kind?

@bukepo

Reply:
> Can we add an optional argument to RESET command to specify the reset kind?

That will break the compatibility between RCP and host. I think we have 2 approaches:
- Add a build flag to choose reset behavior
- Leave the current `RESET` command untouched and add a new `STACK_RESET` command

Reply:
> > Can we add an optional argument to RESET command to specify the reset kind?
> 
> That will break the compatibility between RCP and host. I think we have 2 approaches:

Not, it won't as long as its an optional argument. I think it's fine to add more fields to a Spinel message structure. 

> * Add a build flag to choose reset behavior
> * Leave the current `RESET` command untouched and add a new `STACK_RESET` command

That works for me also.

Reply:
Optional argument looks good. Updated.